File: backend/models/item.go
package models

type Item struct {
	ID             uint         `json:"id" gorm:"primaryKey"`
	Name           string       `json:"name,omitempty"`
	Price          float64      `json:"price,omitempty"`
	ImageName      string       `json:"image_name,omitempty"`
	Description    string       `json:"description,omitempty"`
	ItemCategoryID uint         `json:"item_category_id,omitempty"`
	ItemCategory   ItemCategory `json:"item_category,omitempty" gorm:"foreignKey:ItemCategoryID"`
	Stock          int          `json:"stock,omitempty"`
}

File: frontend/types/item.ts
import { z } from 'zod';

export const ItemSchema = z.object({
  id: z.number(),
  name: z.string(),
  price: z.number(),
  image_name: z.string().optional(),
  description: z.string().optional(),
  item_category_id: z.number().optional(),
  item_category: z.object({
    id: z.number(),
    name: z.string(),
  }).optional(),
  stock: z.number().optional(),
});

export const ItemCategorySchema = z.object({
  id: z.number().int(),
  name: z.string(),
});


export type Item = z.infer<typeof ItemSchema>;
export type ItemCategory = z.infer<typeof ItemCategorySchema>;
